prebid/src/main/java/com/uid2/prebid/UID2Prebid.kt

```diff
@@ -0,0 +1,132 @@
+import com.uid2.UID2ManagerState.Refreshed
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import org.prebid.mobile.ExternalUserId
+import org.prebid.mobile.PrebidMobile
+
+/**
+ * Interface to wrap access to [PrebidMobile]. This is used to improve testability, rather than having the [UID2Prebid]
+ * access it via static methods.
+ */
+internal fun interface PrebidExternalUserIdInteractor {
+    operator fun invoke(ids: List<ExternalUserId>)
+}
+
+/**
+ * Prebid integration that will observe a given [UID2Manager] instance and update Prebid when a new [ExternalUserId] is
+ * available. After creating the instance, a consumer must explicitly call [initialize] for this instance to start
+ * observing changes.
+ */
+public class UID2Prebid internal constructor(
+    private val manager: UID2Manager,
+    private val externalUserIdFactory: () -> List<ExternalUserId>,
+    private val prebidInteractor: PrebidExternalUserIdInteractor,
+    dispatcher: CoroutineDispatcher,
+) {
+
+    // We redirect to the logger owned by the UID2Manager, as it's been configured correctly.
+    private val logger = manager.logger
+
+    /**
+     * Constructor.
+     *
+     * @param manager The [UID2Manager] instance to be observed.
+     * @param externalUserIdFactory A factory that will allow the consumer to add any other [ExternalUserId]s that should
+     *                              also be included, rather than just a single list containing only UID2's instance.
+     */
+    @JvmOverloads
+    public constructor(
+        manager: UID2Manager,
+        externalUserIdFactory: () -> List<ExternalUserId> = { emptyList() },
+    ) : this(
+        manager,
+        externalUserIdFactory,
+        PrebidExternalUserIdInteractor { ids -> PrebidMobile.setExternalUserIds(ids) },
+        Dispatchers.Default,
+    )
+
+    private val scope = CoroutineScope(dispatcher + SupervisorJob())
+
+    /**
+     * Initializes the integration which will start observing the associated [UID2Manager] instance for changes in the
+     * availability of the advertising token. As the token is refreshed, this will automatically update Prebid's list
+     * of ExternalUserIds.
+     */
+    public fun initialize() {
+        // Once the UID2Manager instance has been initialized, we will start observing it for changes.
+        manager.addOnInitializedListener {
+            updateExternalUserId(manager.getAdvertisingToken(), "Initialized")
+            observeIdentityChanges()
+        }
+    }
+
+    /**
+     * Releases this instance, to not be used again.
+     */
+    public fun release() {
+        scope.cancel()
+    }
+
+    /**
+     * Returns the list of UID2 scoped [ExternalUserId]s.
+     */
+    public fun getExternalUserIdList(): List<ExternalUserId> {
```

Comment:
Incase a consumer wants to explicitly update the `ExternalUserId`s on Prebid, they can use this to access our current collection.